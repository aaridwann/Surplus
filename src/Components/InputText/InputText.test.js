import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import AnimatedInputWithLabel from './InputText';

describe('AnimatedInputWithLabel', () => {
	test('should render correctly with default props', () => {
		const { getByText, getByPlaceholderText } = render(
			<AnimatedInputWithLabel label="Label" />
		);

		expect(getByText('Label')).toBeDefined();
		expect(getByPlaceholderText('')).toBeDefined();
	});

	test('should render label with custom text', () => {
		const label = 'Custom Label';
		const { getByText } = render(<AnimatedInputWithLabel label={label} />);

		expect(getByText(label)).toBeDefined();
	});

	test('should update input value correctly', () => {
		const onChangeMock = jest.fn();
		const { getByPlaceholderText } = render(
			<AnimatedInputWithLabel label="Label" onChange={onChangeMock} />
		);
		const input = getByPlaceholderText('');

		fireEvent.changeText(input, 'New Value');

		expect(onChangeMock).toHaveBeenCalledWith('New Value');
	});

	// Add more test cases as needed
});

// Snapshot test
describe('AnimatedInputWithLabel snapshots', () => {
	test('should match the snapshot', () => {
		const { toJSON } = render(<AnimatedInputWithLabel label="Label" />);

		expect(toJSON()).toMatchSnapshot();
	});
});
