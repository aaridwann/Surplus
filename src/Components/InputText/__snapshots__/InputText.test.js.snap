// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedInputWithLabel snapshots should match the snapshot 1`] = `
<View
  style={
    Object {
      "marginBottom": 20,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#50b8a8",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginBottom": 5,
        },
        Object {
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        },
      ]
    }
  >
    Label
  </Text>
  <TextInput
    autoCapitalize="none"
    collapsable={false}
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder=""
    style={
      Object {
        "borderColor": "#ccc",
        "borderWidth": 1,
        "fontSize": 16,
        "height": 40,
        "paddingHorizontal": 10,
      }
    }
  />
</View>
`;
